lib/reverse_sentence.rb

```diff
@@ -1,6 +1,45 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(n)
```

Comment:
So you are pulling out each word and putting them into a new array in reverse order.  This seems a bit overly complicated and it's not in-place.  

How about this pseudocode:

start_index = 0
last_index = length - 1

while start_index is less than last_index
  swap the letters at 1st and last indexes
  last index goes down by 1
  first index goes up by one
end

You can also build a method to do the above for individual words.  So reverse the whole string, then go through the string and reverse each word.